Apps/Laboratory/src/Services/LaboratoryService.cs

```diff
@@ -86,224 +86,163 @@ public class LaboratoryService : ILaboratoryService
         {
-            RequestResult<PublicCovidTestResponse> retVal = new()
-            {
-                ResultStatus = ResultType.Error,
-                ResourcePayload = new PublicCovidTestResponse(),
-            };
-
-            DateOnly dateOfBirth;
-            try
+            if (!DateFormatter.TryParse(dateOfBirthString, "yyyy-MM-dd", out DateTime dateOfBirth))
             {
-                dateOfBirth = DateOnly.ParseExact(dateOfBirthString, "yyyy-MM-dd", CultureInfo.InvariantCulture);
-            }
-            catch (Exception e) when (e is FormatException || e is ArgumentNullException)
-            {
-                retVal.ResultStatus = ResultType.Error;
-                retVal.ResultError = new RequestResultError
-                {
-                    ResultMessage = "Error parsing date of birth",
-                    ErrorCode = ErrorTranslator.InternalError(ErrorType.InvalidState),
-                };
-                return retVal;
+                return RequestResultFactory.Error<PublicCovidTestResponse>(ErrorType.InvalidState, "Error parsing date of birth");
             }
 
-            DateOnly collectionDate;
-            try
+            if (!DateFormatter.TryParse(collectionDateString, "yyyy-MM-dd", out DateTime collectionDate))
             {
-                collectionDate = DateOnly.ParseExact(collectionDateString, "yyyy-MM-dd", CultureInfo.InvariantCulture);
-            }
-            catch (Exception e) when (e is FormatException || e is ArgumentNullException)
-            {
-                retVal.ResultStatus = ResultType.Error;
-                retVal.ResultError = new RequestResultError
-                {
-                    ResultMessage = "Error parsing collection date",
-                    ErrorCode = ErrorTranslator.InternalError(ErrorType.InvalidState),
-                };
-                return retVal;
+                return RequestResultFactory.Error<PublicCovidTestResponse>(ErrorType.InvalidState, "Error parsing collection date");
             }
 
             if (!PhnValidator.IsValid(phn))
             {
-                retVal.ResultStatus = ResultType.Error;
-                retVal.ResultError = new RequestResultError
-                {
-                    ResultMessage = "Error parsing phn",
-                    ErrorCode = ErrorTranslator.InternalError(ErrorType.InvalidState),
-                };
-                return retVal;
+                return RequestResultFactory.Error<PublicCovidTestResponse>(ErrorType.InvalidState, "Error parsing phn");
             }
 
             string? accessToken = this.authenticationDelegate.AuthenticateAsSystem(this.tokenUri, this.tokenRequest).AccessToken;
-            if (string.IsNullOrEmpty(accessToken))
+            if (accessToken == null)
             {
-                this.logger.LogCritical(IsNullOrEmptyTokenErrorMessage);
```

Comment:
Why was this log statement removed?

Reply:
made it consistent with the other checks in the other methods